paddle/fluid/framework/tensor_util.cc

```diff
@@ -495,6 +495,127 @@ void TensorFromStream(std::istream& is, Tensor* tensor,
   }
 }
 
+// get tensor data point by DLDataType
+void* GetDstPtrByDLDataType(DLDataType type, framework::Tensor* dst,
+                            const platform::CPUPlace& dst_place) {
+  if (type.lanes > 1) return nullptr;  // vector types not currently supported
+
+  switch (type.bits) {
+    case 8:
+      return (type.code == kDLInt)
+                 ? static_cast<void*>(dst->mutable_data<int8_t>(dst_place))
+                 : (type.code == kDLUInt)
+                       ? static_cast<void*>(
+                             dst->mutable_data<uint8_t>(dst_place))
+                       : nullptr;
+    case 16:
+      return (type.code == kDLInt)
+                 ? static_cast<void*>(dst->mutable_data<int16_t>(dst_place))
+                 : (type.code == kDLFloat)
+                       ? static_cast<void*>(
+                             dst->mutable_data<paddle::platform::float16>(
+                                 dst_place))
+                       : nullptr;
+    case 32:
+      return (type.code == kDLInt)
+                 ? static_cast<void*>(dst->mutable_data<int32_t>(dst_place))
+                 : (type.code == kDLFloat)
+                       ? static_cast<void*>(dst->mutable_data<float>(dst_place))
+                       : nullptr;
+    case 64:
+      return (type.code == kDLInt)
+                 ? static_cast<void*>(dst->mutable_data<int64_t>(dst_place))
+                 : (type.code == kDLFloat)
+                       ? static_cast<void*>(
+                             dst->mutable_data<double>(dst_place))
+                       : nullptr;
+    default:
+      break;
+  }
+
+  return nullptr;
+}
+
+#ifdef PADDLE_WITH_CUDA
+void* GetDstPtrByDLDataType(DLDataType type, framework::Tensor* dst,
```

Comment:
You can simplify CPU and GPU version `GetDstPtrByDLDataType` to be one function with declaration like:

`void* GetDstPtrByDLDataType(DLDataType type, framework::Tensor* dst, const platform::Place &dst_place);`

Reply:
Good Idea!
Done!